zerver/views/user_groups.py

```diff
@@ -0,0 +1,102 @@
+from django.http import HttpResponse, HttpRequest
+from django.utils.translation import ugettext as _
+
+from typing import List, Text
+from zerver.models import UserProfile, UserGroup
+
+from zerver.context_processors import get_realm_from_request
+from zerver.lib.request import has_request_variables, REQ
+from zerver.lib.response import json_success, json_error
+from zerver.lib.validator import check_list, check_string, check_int, \
+    check_short_string
+
+from zerver.lib.actions import check_add_user_group, do_update_user_group_name, \
+    do_update_user_group_description
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.user_groups import get_user_group_by_id, \
+    get_memberships_of_users
+from zerver.lib.actions import bulk_add_members_to_user_group, \
+    remove_members_from_user_group
+from zerver.views.streams import compose_views, FuncKwargPair
+
+def check_get_user_group(user_group_id):
+    # type: (int) -> UserGroup
+    try:
+        return get_user_group_by_id(user_group_id)
+    except UserGroup.DoesNotExist:
+        raise JsonableError(_("Invalid user group"))
+
+@has_request_variables
+def add_user_group(request, user_profile,
+                   name=REQ(),
+                   members=REQ(validator=check_list(check_int), default=[]),
+                   description=REQ()):
+    # type: (HttpRequest, UserProfile, Text, List[int], Text) -> HttpResponse
+    realm = get_realm_from_request(request)
```

Comment:
use `user_profile.realm`; `get_realm_from_request` isn't for this sort of thing.